### ISSUE TITLE
Processing of CSS of changes

### ISSUE THREAD

**Issue by MrChebik** [TomasMikula/RichTextFX#586]
I have the listener:
https://github.com/TomasMikula/RichTextFX/blob/master/richtextfx-demos/src/main/java/org/fxmisc/richtext/demo/JavaKeywordsAsync.java#L93-L106

After compute highlighting, I run checker syntax, and use `setStyleClass` method to `CodeArea` if an error or an assign operation have met. But when I do this, listener of `CodeArea` begin do task again and again. Whether I can make so that the listener didn't pay attention on CSS changes?

**Comment by JordanMartinez** [comment]
Sorry, could you describe your problem again? I am finding it difficult to understand you.

**Comment by MrChebik** [comment]
Yes, when I typed anything at `CodeArea`, listener above compute highlighting.
I add a method, that compute correct syntax of code, and run after highlight:

```
private void applyHighlighting(StyleSpans<Collection<String>> highlighting) {
        this.setStyleSpans(0, highlighting);
        syntax.compute(this);
}
```

At the `syntax.compute(this)`, If this method meet an error at the source code, I add a class to `CodeArea` with `from` and `to` position: `codePlace.setStyleClass(from, to, "error");`

After that, the listener see this changes CSS, and run again method described above: `applyHighlighting`.
I don't want to reload the computing highlight if `CodeArea` has any changes with CSS classes/positions.

**Comment by JordanMartinez** [comment]
Oh ok. Here's what's going on and why your problem is occurring:
1. The area's text or style is changed, so an event is emitted
3. After 500 ms of inactivity, the `computeHighlighting` method (which returns the spans of styles to apply to the area's text) is called
4. Once that task is finished, it applies those style spans to the area via `setStyleSpans`.
5. You call `syntax.compute`, which then calls `setStyleClass`.
6. `setStyleClass` ultimately calls `area.replace`, which triggers a change in the area's underlying document. Go to Step 1 and loop infinitely

You need to run your syntax computation method in `computeHighlighting`, not in `applyHighlighting`, and somehow integrate it into the returned `StyleSpans` object.

**Comment by MrChebik** [comment]
Thank you. It will be difficult to integrate with `StyleSpans`.
In version 0.7-M5, I can't work with this object, there is some equivalent?

**Comment by JordanMartinez** [comment]
Try `StyleSpansBuilder`

**Comment by MrChebik** [comment]
Sorry, I haven't noticed that packages have changed.
I resolved my problem with tricks.

1. For CSS changes, at the `filter`, I added condition for compare last text and current text at the `CodeArea`
2. At the Syntax highlighting, I used `Platform.runLater` and received blinking highlight of syntax. For resolve this problem, I created new `CodeArea` for CSS operations at the end of computing, I get CSS from here and set at the "drawing" `CodeArea`.

Thank you very much, I didn't expect so fast response!
